raiden/utils/migrations/v16_to_v17.py

```diff
@@ -0,0 +1,53 @@
+from raiden.exceptions import InvalidDBData
+from raiden.storage.restore import rebuild_chain_state
+from raiden.storage.sqlite import SQLiteStorage
+
+
+def _get_latest_state_change(storage: SQLiteStorage):
+    cursor = storage.conn.cursor()
+    sql = (
+        f'SELECT identifier '
+        f'FROM state_changes '
+        f'ORDER BY identifier '
+        f'DESC LIMIT 1'
+    )
+    cursor.execute(sql)
+    try:
+        row = cursor.fetchone()
+        if not row:
+            return 0
+        return row[0]
+    except AttributeError:
+        raise InvalidDBData(
+            'Your local database is corrupt. Bailing ...',
+        )
+
+
+def _remove_snapshots(storage: SQLiteStorage):
+    storage.remove_snapshots()
```

Comment:
Upgrading them on-demand would mean complexity in the codebase of the state-machine (migrate on access or something of that sort). Otherwise, what's the alternative in your opinion? the reason i am doing this is because i don't want the node to end up loading an old snapshot and applying state changes from there, that's the reason i am rebuilding it to guarantee that any snapshot will have already been migrated. This would have been easier if a code change was applied on a state_change or an event because i would query them from the database and change on inplace... but since the code change happened in the state (which is built based on state changes), this was my only option that i could think of.